src/Package/Impl/DataInspect/DataImport/ImportDataWindow.xaml.cs

```diff
@@ -309,11 +256,135 @@ class DataFramePreviewRowItem {
         }
 
         private void ComboBox_SelectionChanged(object sender, SelectionChangedEventArgs e) {
-            PreviewAsync().DoNotWait();
+            PreviewContent();
         }
 
         private void HeaderCheckBox_Changed(object sender, RoutedEventArgs e) {
-            PreviewAsync().DoNotWait();
+            PreviewContent();
+        }
+
+        protected override void OnClosed(EventArgs e) {
+            DeleteTempFile();
+            base.OnClosed(e);
+        }
+
+        private void PreviewFileContent(string file, int codePage) {
+            Encoding encoding = Encoding.GetEncoding(codePage);
+            string text = ReadFilePreview(file, encoding);
+            InputFilePreview.Text = text;
+        }
+
+        private Task ConvertToUtf8(string file, int codePage, int maxLines = Int32.MaxValue) {
+            DeleteTempFile();
+            bool reportProgress = maxLines == Int32.MaxValue;
+
+            // Convert file to UTF-8 so it can be properly imported
+            return Task.Run(async () => {
+                try {
```

Comment:
If you move the body of the `try` block into separated method, you can simply call `await TaskUtilities.SwitchToBackgroundThread();` at its beginning, and also you can simply call `OnError(ex.Message))` instead of `VsAppShell.Current.DispatchOnUIThread(() => OnError(ex.Message));`